internal/server/server.go

```diff
@@ -21,6 +23,60 @@ import (
 
 type Handlers struct{}
 
+func OsbuildClient() (*cloudapi.Client, error) {
+	socket, ok := os.LookupEnv("OSBUILD_SERVICE")
+	if !ok {
+		socket = "http://127.0.0.1:80"
+	}
+	Log.Infoln("Creating osbuildclient with socket %s", socket)
+	return cloudapi.NewClient(socket, ConfigureClient)
+}
+
+func ConfigureClient(client *cloudapi.Client) error {
+	// set up client certificate authentication
+	if (strings.HasPrefix(client.Server, "https")) {
+		certFile, ok := os.LookupEnv("OSBUILD_CERT_PATH")
+		if !ok {
+			return fmt.Errorf("Couldn't find cert path, is OSBUILD_CERT_PATH set?")
+		}
+		keyFile, ok := os.LookupEnv("OSBUILD_KEY_PATH")
+		if !ok {
+			return fmt.Errorf("Couldn't find key path, is OSBUILD_KEY_PATH set?")
+		}
+		caFile, hasCA := os.LookupEnv("OSBUILD_CA_PATH")
+
+		// Load client cert
+		cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+		if err != nil {
+			return err
+		}
+
+		var tlsConfig *tls.Config
+		if hasCA {
+			caCert, err := ioutil.ReadFile(caFile)
+			if err != nil {
+				return err
+			}
+			caCertPool := x509.NewCertPool()
+			caCertPool.AppendCertsFromPEM(caCert)
+			tlsConfig = &tls.Config{
+				Certificates: []tls.Certificate{cert},
+				RootCAs:      caCertPool,
+			}
+		} else {
```

Comment:
Don't think we need to / should support the case where the CA is not provided. @larskarlitski?

Reply:
Isn't this useful for local development?

Reply:
In composer we just had a make target to generate certs, but this is really a nitpick and something we could change in the future, so if you are using it like this for development, then keep it as is for now.